metrics_sdk/lib/opentelemetry/sdk/metrics/state/metric_store.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module State
+        # @api private
+        #
+        # The MetricStore module provides SDK internal functionality that is not a part of the
+        # public API.
+        class MetricStore
+          def initialize
+            @mutex = Mutex.new
+            @epoch_start_time = now_in_nano
```

Comment:
An `epoch` from start_time to end_time is basically the  duration between calls to `collect`, is that right?